providers/implementations/keymgmt/ec_kmgmt.c

```diff
@@ -743,6 +743,15 @@ int common_get_params(void *key, OSSL_PARAM params[], int sm2)
             ERR_raise(ERR_LIB_PROV, PROV_R_NOT_A_PUBLIC_KEY);
             goto err;
         }
+
+        p->return_size = EC_POINT_point2oct(ecg, ecp,
+                                            POINT_CONVERSION_UNCOMPRESSED,
+                                            NULL, 0, bnctx);
+        /* Check input buffer size */
+        if (p->data != NULL && p->data_size < p->return_size) {
```

Comment:
When p->data is NULL the second call to EC_POINT_point2oct() is redundant..

Reply:
Yes It is possible. How about below code:
```
        p->return_size = EC_POINT_point2oct(ecg, ecp,
                                            POINT_CONVERSION_UNCOMPRESSED,
                                            NULL, 0, bnctx);
        /* Check input buffer size */
        if (p->return_size == 0
            || (p->data != NULL && p->data_size < p->return_size)) {
            /* Buffer too small or return size is 0, return error value 0. */
            goto err;
        }

        if (p->data != NULL) {
            p->return_size = EC_POINT_point2oct(ecg, ecp,
                                            POINT_CONVERSION_UNCOMPRESSED,
                                            p->data, p->return_size, bnctx);
            if (p->return_size == 0)
                goto err;
        }
```